app/views/user/swaps/new.html.haml

```diff
@@ -5,8 +5,20 @@
     = render partial: "user/swaps/double_check_constituency", locals: { swap_with: @swap_with }
     %p.text-center
       Are you sure you would like to swap your vote with #{@swap_with.redacted_name}?
+
     %p.text-center
-      = link_to "Yes", user_swap_path(user_id: @swap_with.id), method: :post, class: "btn btn-primary"
+      = form_tag user_swap_path, class: "form-inline" do
+
+        = hidden_field_tag 'user_id', @swap_with.id
+        %p.text-center.small
+          Please check the box if you are willing to share your
```

Comment:
Can we use a BS form here, put the checkbox before the label and link the label to the checkbox (so the click area is a bit bigger)? I think I might miss this if I wasn't careful.

Reply:
I can look at this if you want?

Reply:
Please  :)